File: lang/python/algorithms/algos/addfractions.py
def AddFractions(fr1,fr2):
    a = str(fr1).split('/')
    b = str(fr2).split('/')
    nr1 = int(a[0])
    dr1 = int(a[1])
    nr2 = int(b[0])
    dr2 = int(b[1])

    lcm = getLCM(dr1,dr2)
    print(lcm)

    nr = lcm/dr1 * nr1 + lcm/dr2 * nr2
    gcd = getGCD(nr,lcm)
    print("{0}/{1}".format(nr/gcd,lcm/gcd))



def getLCM(num1,num2):
    a = num1
    b = num2
    if(num2>num1):
        a = num2
        b = num1

    for x in range(1,b):
        if ((a*x)%b==0):
            return a*x

    return num1*num2

def getGCD(num1,num2):
    if(num1==0):
        return num2
    if(num2==0):
        return num1

    if(num1==num2):
        return num1

    if(num1>num2):
        return getGCD(num1-num2,num2)

    return getGCD(num1,num2-num1)

File: lang/CSharp/algorithms/Algos/AddFractions.cs
using System;
using algorithms.Common;

namespace algorithms.Algos
{
    public class AddFractions : IRunnable
    {
        public AddFractions(IProjIO io)
        {
            IO = io;
        }
        public IProjIO IO{get;private set;}


        public void Run()
        {
            IO.WriteLine("Enter Fraction 1");
            string[] fr1 = IO.Read().Split('/');
            int nr1 = int.Parse(fr1[0]);
            int dr1 = int.Parse(fr1[1]);
            IO.WriteLine("Enter Fraction 2");
            string[] fr2 = IO.Read().Split('/');
            int nr2 = int.Parse(fr2[0]);
            int dr2 = int.Parse(fr2[1]);

            IO.Write(SumFractions(nr1,dr1,nr2,dr2));
        }

        private string SumFractions(int nr1, int dr1, int nr2, int dr2)
        {
            int lcm = GetLCM(dr1,dr2);
            IO.WriteLine(lcm.ToString());
            int nr = (lcm/dr1 * nr1) + (lcm/dr2 * nr2);
            int gcd = GetGCD(nr,lcm);
            return $"{nr/gcd}/{lcm/gcd}";
        }

        private int GetLCM(int dr1, int dr2)
        {
            //IO.WriteLine($"{dr1} and {dr2}");
            int num1 = dr1;
            int num2 = dr2;
            if(dr2>dr1){
                num1 = dr2;
                num2 = dr1;
            }

            for(int i=1;i<num2;i++){
                if((num1*i)%num2==0){
                    return num1*i;
                }
            }

            return num1*num2;
        }

     private int GetGCD(int num1,int num2){
         if(num1==0)
            return num2;
        if(num2==0)
            return num1;

        if(num1==num2)
            return num1;

        if(num1>num2)
            return GetGCD(num1-num2,num2);

        return GetGCD(num1,num2-num1);

     }
    }
}